lib/pgBackRest/Storage/Base.pm

```diff
@@ -92,47 +114,62 @@ sub copy
         $strOperation,
         $xSourceFile,
         $xDestinationFile,
+        $bSourceOpen,
     ) =
         logDebugParam
         (
             __PACKAGE__ . '->copy', \@_,
             {name => 'xSourceFile', required => false},
-            {name => 'xDestinationFile', required => false},
+            {name => 'xDestinationFile'},
+            {name => 'bSourceOpen', optional => true, default => false},
         );
 
-    # Was the file copied?
-    my $bResult = false;
+    # Is source/destination an IO object or a file expression?
+    my $oSourceFileIo = defined($xSourceFile) ? (ref($xSourceFile) ? $xSourceFile : $self->openRead($xSourceFile)) : undef;
 
-    # Is source an IO object or a file expression?
-    my $oSourceFileIo =
-        defined($xSourceFile) ?
-        (ref($xSourceFile) ? $xSourceFile : $self->openRead($self->pathGet($xSourceFile))) : undef;
```

Comment:
The C code runs pathGet() internally so generally we don't need it.  Having it in hashSize() seems like an oversight so I have removed it and we'll see how the tests go.  In all other places this is pasted Posix code which still needs this translation since it is not calling into the C code.

This is the Frankendriver.

Reply:
Test fine ran after this instance was removed.